src/main/java/com/urlparser/entity/LinksFromUrl.java

```diff
@@ -0,0 +1,46 @@
+package com.urlparser.entity;
+
+import com.urlparser.model.Links;
+
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+public class LinksFromUrl implements Links {
+
+  private Document doc;
+  List<URL> listedLinks = new ArrayList<>();
+
+  public LinksFromUrl(String url) {
+    try {
+      doc = Jsoup.connect(url).get();
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+    listed();
+  }
+
+  public void listed() {
```

Comment:
@undsett This method does nothing but changes the internal state of the object. Usually, state introduces additional complication to the code and should be avoided till there is no other ways(state machine tasks, but this is not the case here)
In this particular case `listed` method is confusing, not only because it introduces state regulation, but also it is considered to be called before iterating the object, why?